test/apps/iodemo/io_demo.cc

```diff
@@ -1646,7 +1648,11 @@ class DemoClient : public P2pDemoCommon {
 
     void disconnect_server(size_t server_index, const char *reason) {
         server_info_t& server_info = _server_info[server_index];
-        bool disconnecting         = server_info.conn->is_disconnecting();
+        if (server_info.conn == NULL) {
```

Comment:
move if to disconnect_servers
here - assert conn non NULL

Reply:
Done